pkg/odo/cli/service/create.go

```diff
@@ -270,43 +282,34 @@ func (o *ServiceCreateOptions) Validate() (err error) {
 				return err
 			}
 
-			var almExamples []map[string]interface{}
-			val, ok := csv.Annotations["alm-examples"]
-			if ok {
-				err = json.Unmarshal([]byte(val), &almExamples)
-				if err != nil {
-					return errors.Wrap(err, "unable to unmarshal alm-examples")
-				}
-			} else {
-				// There's no alm examples in the CSV's definition
-				return fmt.Errorf("Could not find alm-examples in operator's definition.\nPlease provide a file containing yaml specification to start the %s service from %s operator", o.CustomResource, o.ServiceName)
-			}
-
-			almExample, err := getAlmExample(almExamples, o.CustomResource, o.ServiceType)
+			almExample, err := svc.GetAlmExample(csv, o.CustomResource, o.ServiceType)
 			if err != nil {
 				return err
 			}
+
 			o.CustomResourceDefinition = almExample
-			o.group, o.version = groupVersionALMExample(almExample)
-			o.resource = resourceFromCSV(csv, o.CustomResource)
-			o.ServiceName, err = serviceNameFromCRD(o.CustomResourceDefinition, o.ServiceName)
-			return err
-		} else {
-			// prevent user from executing `odo service create <operator-name>`
-			// because the correct way is to execute `odo service
-			// <operator-name> --crd <crd-name>`
-			csvs, err := o.KClient.GetClusterServiceVersionList()
+
+			o.group, o.version, o.resource, err = svc.GetGVRFromOperator(csv, o.CustomResource)
 			if err != nil {
 				return err
 			}
 
-			for _, csv := range csvs.Items {
-				if csv.Name == o.ServiceType {
-					// this is satisfied if user has specified operator but not
-					// a CRD name
-					return errors.New("Please specify service name along with the operator name")
+			if o.ServiceName != "" {
```

Comment:
This is the main part from where we ensure that user provided name is used to create the service.